lib/hashes.md

```diff
@@ -5,40 +5,43 @@ instructor_notes: Feel free to re-organize the headings (or add/remove headings)
 
 # What is a Hash?
 
-Replace me with your response...
+A Hash is a collection of key-value pairs very similar to a dictionary. It are similar to Arrays, but where an Array uses integers as its index, a Hash allows you to use any object type.
 
 # What are some examples of information that would be Hashes as opposed to some other data type?
 
-Replace me with your response...
+You would use Hashes when you want to be able to use different elements/categories to index your values. 
 
 # How are Hashes and Arrays similar? How are they different?
 
-Replace me with your response...
+They are similar since they are both use to store a collection of data.
+
+They are different because arrays store the data in a index-numbered list whereas hashes store data with key-value pairs by using strings, numbers, or symbols. 
 
 # How do you retrieve a particular value from a Hash?
 
-Replace me with your response...
+By calling the values key. For example hash = {"key" => "value", "another_key" => "another value"}
 
 # How do you add information to a Hash?
 
-Replace me with your response...
+my_hash_name[new_key_name]= new_value
 
 # How would you perform an operation on every element inside a Hash?
 
-Replace me with your response...
+my_hash.each do |key, value|
+    my_hash[key] = new_value (whatever edit I want to make for every value)
 
 # How would you change the value of a particular element in a Hash?
 
-Replace me with your response...
+my_hash_name[particular_key]= new_value
 
 # How do you delete an element from a Hash?
 
-Replace me with your response...
+my_hash_name.delete(element)
```

Comment:
by calling the name of the key. I should have replaced the word element with key.